### PR TITLE
feat: add Maester adapter for M365/Azure security testing

### PR DESCRIPTION
**Pull request by riskandreason**
## Summary

- Add complete Maester adapter package (adapter, parser, declarative mappings, fixture data) following the ScubaGear reference pattern
- Support Maester's multi-framework test output: CIS M365, CISA SCuBA, EIDSCA, ORCA, and MT community tests with correct ID formats, 5 result statuses, and Block-based category mapping
- Update CIS M365 cross-reference YAML with correct Maester CISA overlap IDs replacing fabricated MT.XXXX placeholders
- Register `maester` entry point in pyproject.toml

## Test Plan

- [x] 27 mapping tests (severity, block-category, dedup key coverage)
- [x] 19 parser tests (all ID formats, statuses, null ResultDetail handling, edge cases)
- [x] 29 conformance tests (21 inherited suite + 8 Maester-specific)
- [x] Cross-reference consistency verified (7 CISA dedup keys match crossref YAML)
- [x] Full suite: 1265 passed, 3 skipped, 89.53% coverage
- [x] All files under 400-line limit
- [x] All pre-commit hooks pass (ruff, pyright, detect-secrets)

### CODE REVIEW

File: src/gxassessms/adapters/maester/adapter.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+        - TestResults-{timestamp}.md (markdown summary)
+
+        There is NO -OutputFormat parameter. -OutputFolder generates all three.
+        """
+        from gxassessms.core.config.datetime_utils import utc_now
+
+        tool_config = config.tools.get("maester", {})
+        timeout = tool_config.get("timeout", 600)
+        output_dir = Path(config.engagement_dir) / "raw" / "maester"
+        output_dir.mkdir(parents=True, exist_ok=True)
```
[the remark this answers is not in the document]
Comment (riskandreason) [reply src/gxassessms/adapters/maester/adapter.py]: Dismissed -- matches the ScubaGear reference adapter pattern. If this is a concern it should be addressed project-wide.

File: src/gxassessms/adapters/maester/mappings.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+# Empty string is possible in real output and defaults to INFO.
+# ---------------------------------------------------------------------------
+
+SEVERITY_MAP: dict[str, Severity] = {
+    "Critical": Severity.CRITICAL,
+    "High": Severity.HIGH,
+    "Medium": Severity.MEDIUM,
+    "Low": Severity.LOW,
+    "Info": Severity.INFO,
+    "": Severity.INFO,
```
[the remark this answers is not in the document]
Comment (riskandreason) [reply src/gxassessms/adapters/maester/mappings.py]: Fixed in 0e46708. `SEVERITY_MAP` now uses `(native_severity, FindingStatus)` tuple keys matching the ScubaGear pattern.
